sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/schema/JsonSchemaManager.java

```diff
@@ -0,0 +1,190 @@
+            });
+  }
+
+  /**
+   * Validates a JSON document against a given table's schema
+   *
+   * @param table the table that has a schema
+   * @param document the document, as stringified JSON
+   * @return a Uni with Boolean detailing whether or not the document complies with the schema.
+   */
+  public Uni<Boolean> validateJsonDocument(Uni<Schema.CqlTable> table, String document) {
+    return table
+        .onItem()
+        .ifNotNull()
+        .transform(
+            t -> {
+              String comment = t.getOptionsMap().getOrDefault("comment", null);
+              if (comment == null) {
+                // If there is no valid schema, then the document is valid
+                return true;
+              }
+              JsonNode jsonSchema;
+              try {
+                jsonSchema = objectMapper.readTree(comment);
+              } catch (JsonProcessingException e) {
+                // If there is no valid schema, then the document is valid
+                return true;
+              }
+
+              try {
+                validate(jsonSchema.get("schema"), document);
+              } catch (ProcessingException e) {
+                return true;
+              }
+              return true;
+            });
+  }
+
+  /**
+   * Validates a JSON document against a given table's schema
+   *
+   * @param table the table that has a schema
+   * @param document the document, as JsonNode
+   * @return a Uni with Boolean detailing whether or not the document complies with the schema.
+   */
+  public Uni<Boolean> validateJsonDocument(Uni<Schema.CqlTable> table, JsonNode document) {
+    return table
+        .onItem()
+        .ifNotNull()
+        .transform(
+            t -> {
+              String comment = t.getOptionsMap().getOrDefault("comment", null);
+              if (comment == null) {
+                // If there is no valid schema, then the document is valid
+                return true;
+              }
+              JsonNode jsonSchema;
+              try {
+                jsonSchema = objectMapper.readTree(comment);
+              } catch (JsonProcessingException e) {
+                // If there is no valid schema, then the document is valid
+                return true;
+              }
+
+              try {
+                validate(jsonSchema.get("schema"), document);
+              } catch (ProcessingException e) {
+                return true;
+              }
+              return true;
+            });
+  }
+
+  private void validate(JsonNode schema, String value) throws ProcessingException {
+    final JsonNode tree;
+    try {
+      tree = objectMapper.readTree(value);
+    } catch (JsonProcessingException e) {
+      throw new ErrorCodeRuntimeException(
+          ErrorCode.DOCS_API_INVALID_JSON_VALUE, "Malformed JSON object found during read: " + e);
+    }
+    validate(schema, tree);
+  }
+
+  private void validate(JsonNode schema, JsonNode jsonValue) throws ProcessingException {
```

Comment:
so what's with the processing exception, should we catch that in `validateJsonDocument` and report with some kind of the `ErrorCode`? Or is the intention to end up in the internal server error? That would be fine as well..

Reply:
Hmm this probably should bubble up into some ErrorCode...I'll change it

Reply:
I guess we should check on V1 here!